graph/src/data/schema.rs

```diff
@@ -606,6 +714,192 @@ impl Schema {
             })
     }
 
+    fn validate_fulltext_directives(&self) -> Vec<SchemaValidationError> {
+        self.subgraph_schema_object_type()
+            .map_or(vec![], |subgraph_schema_type| {
+                subgraph_schema_type
+                    .directives
+                    .iter()
+                    .filter(|directives| directives.name.eq("fulltext"))
+                    .fold(vec![], |mut errors, fulltext| {
+                        self.validate_fulltext_directive_name(fulltext)
+                            .into_iter()
+                            .for_each(|err| errors.push(err));
+                        self.validate_fulltext_directive_language(fulltext)
+                            .into_iter()
+                            .for_each(|err| errors.push(err));
+                        self.validate_fulltext_directive_algorithm(fulltext)
+                            .into_iter()
+                            .for_each(|err| errors.push(err));
+                        self.validate_fulltext_directive_includes(fulltext)
+                            .into_iter()
+                            .for_each(|err| errors.push(err));
+                        errors
+                    })
+            })
+    }
+
+    fn validate_fulltext_directive_name(&self, fulltext: &Directive) -> Vec<SchemaValidationError> {
+        let name_value = fulltext
+            .arguments
+            .iter()
+            .find(|(key, _)| key.eq("name"))
+            .map(|(_, value)| value);
+        let name = match name_value {
+            Some(Value::String(name)) => name,
+            _ => return vec![SchemaValidationError::FulltextNameUndefined],
+        };
+
+        // Validate that fulltext.name does not overlap with any of the types
+        // TODO: Validate that each fulltext directive has a distinct name
+        let mut all_types = self
+            .document
+            .get_object_and_interface_type_fields()
+            .into_iter()
+            .map(|(name, _)| name.clone())
+            .collect::<Vec<String>>();
+        let mut imported_types = self
+            .imported_types()
+            .into_iter()
+            .map(|(imported_type, _)| match imported_type {
+                ImportedType::Name(name) => name.clone(),
+                ImportedType::NameAs(_, az) => az.clone(),
+            })
+            .collect::<Vec<String>>();
+        all_types.append(&mut imported_types);
+
+        if let Some(_) = all_types.iter().find(|typ| typ.eq(&name)) {
```

Comment:
I think this check doesn't really catch conflicts. The fulltext directive name defines the query field that will be generated. The type names _map_ to query fields but they are query fields themselves. So unless I'm missing something, this is comparing 🍎s to 🍊s.

Reply:
Ahh totally, removing.